File: 0205 isomorphic-strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
     if(s.length()!=t.length()) return false;
     int[] a=new int[256];
     int[] b=new int[256];

     for(int i=0;i<s.length();i++){
        char c=s.charAt(i);
        char d=t.charAt(i);
        if(a[c]!=b[d]) return false;
        a[c]=i+1;
        b[d]=i+1;
     }
     return true;
    
    }
}

File: 0205 isomorphic-strings/solution.rs
impl Solution {
    pub fn is_isomorphic(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }

        let mut a = [-1; 256];
        let mut b = [-1; 256];

        for (i, (cs, ct)) in s.chars().zip(t.chars()).enumerate() {
            let si = cs as usize;
            let ti = ct as usize;

            if a[si] != b[ti] {
                return false;
            }

            a[si] = i as i32;
            b[ti] = i as i32;
        }

        true
    }
}

